src/handlers/secret.rs

```diff
@@ -0,0 +1,182 @@
+            true => Some(entity.clue2),
+            false => None,
+        },
+        clue3: match entity.guess_attempts >= attempt_rule.clue3_attempts {
+            true => Some(entity.clue3),
+            false => None,
+        },
+        secret: entity.guessed_secret,
+        guesser: entity.guesser,
+    }
+}
+
+#[post("/secrets", format = "json", data = "<secret>")]
+pub async fn create_secret_handler(
+    storage: &State<Storage>,
+    attempt_rule: &State<AttemptCountRule>,
+    secret: Json<NewSecret>,
+) -> Result<Json<Secret>, status::Custom<Json<Value>>> {
+    let created_secret = storage
+        .create_secret(secret.into_inner())
+        .await;
+
+    let secret = match created_secret {
+        Ok(secret) => secret,
+        Err(_) => return Err(status::Custom(Status::InternalServerError, Json(json!({"error": "Internal Server Error."}))))
+    };
+
+    let processed_secret = process_secret(*attempt_rule.inner(), secret);
+
+    debug!("Create Secret Handler executed successfully.");
+
+    Ok(Json(processed_secret))
+}
+
+#[get("/secrets/<id>")]
+pub async fn get_secret_handler(
+    storage: &State<Storage>,
+    attempt_rule: &State<AttemptCountRule>,
+    id: &str,
+) -> Result<Json<Secret>, status::Custom<Json<Value>>> {
+    let secret_id = match Uuid::parse_str(id) {
+        Ok(id) => id,
+        Err(_) => return Err(status::Custom(Status::BadRequest, Json(json!({"error": "Invalid Secret ID."}))))
+    };
+
+    let storage_secret = storage.get_secret_entity(secret_id).await;
+
+    let secret = match storage_secret {
+        Ok(secret) => secret,
+        Err(AppError::NotFound) => {
+            return Err(status::Custom(
+                Status::NotFound,
+                Json(json!({"error": "Secret not found."})),
+            ));
+        }
+        Err(_) => return Err(status::Custom(Status::InternalServerError, Json(json!({"error": "Internal Server Error."})))),
+    };
+
+    let processed_secret = process_secret(*attempt_rule.inner(), secret);
+
+    debug!("Get Secret Handler executed successfully.");
+
+    Ok(Json(processed_secret))
+}
+
+#[get("/secrets")]
+pub async fn get_all_unguessed_secrets_handler(
+    storage: &State<Storage>,
+    attempt_rule: &State<AttemptCountRule>,
+) -> Result<Json<AllSecretsResponse>, status::Custom<Json<Value>>> {
+    let storage_secrets = storage
+        .get_all_unguessed_secrets()
+        .await;
+
+    let secret_entities = match storage_secrets {
+        Ok(secrets) => secrets,
+        Err(_) => return Err(status::Custom(Status::InternalServerError, Json(json!({"error": "Internal Server Error."}))))
+    };
+
+    let secrets: Vec<Secret> = secret_entities.into_iter().map(|secret_entity| process_secret(*attempt_rule.inner(), secret_entity)).collect();
+
+    let response = AllSecretsResponse {
+        secrets: secrets.clone(),
+    };
+
+    debug!("Get All Secrets Handler executed successfully.");
```

Comment:
Here it's different, I don't think we should append the entire array of secrets returned. Just saying that we returned them is fine for a GET ALL endpoint

Reply:
For this Response, I switched to the simple Json<Vec<Secret>>, see the code below:
```
let secrets: Vec<Secret> = secret_entities
        .into_iter()
        .map(|secret_entity| process_secret(*attempt_rule.inner(), secret_entity))
        .collect();

debug!("Get All Secrets Handler executed successfully and returned {} items.", secrets.len());
    
Ok(Json(secrets))
```